egs/wsj/s5/steps/nnet3/chain2/train.sh

```diff
@@ -0,0 +1,283 @@
+. <(head -n 1 $dir/schedule.txt)
+langs=$(awk '/^langs/ { $1=""; print; }' <$dir/init/info.txt | tail -1)
+num_langs=$(echo $langs | wc -w)
+
+mkdir -p $dir/log
+
+# Copy models with initial learning rate and dropout options from $dir/init to $dir/0
+#for lang in $langs; do
+if [ $stage -le -1 ]; then
+  # run.pl $dir/log/init_model_default.log \
+  #     nnet3-am-copy  --learning-rate=$lrate $dropout_opt $dir/init/default.mdl $dir/0.mdl
+  echo "$0: Copying transition model"
+  if [ $num_langs -eq 1 ]; then
+      echo "$0: Num langs is 1"
+      cp $dir/init/default.raw $dir/0.raw
+      if [ -f $dir/init/default_trans.mdl ]; then
+          cp $dir/init/default_trans.mdl $dir/0_trans.mdl 
+      fi
+  else
+      echo "$0: Num langs is $num_langs"
+      cp $dir/init/multi.raw $dir/0.raw
+  fi
+fi
+
+
+l2_regularize_opt=""
+if [ ! -z $l2_regularize ]; then
+    l2_regularize_opt="--l2-regularize=$l2_regularize"
+fi
+
+x=0
+if [ $stage -gt $x ]; then x=$stage; fi
+
+[ $max_models_combine -gt $[num_iters/2] ] && max_models_combine=$[num_iters/2];
+combine_start_iter=$[num_iters+1-max_models_combine]
+
+while [ $x -lt $num_iters ]; do
+  # Source some variables fromm schedule.txt.  The effect will be something
+  # like the following:
+  # iter=0; num_jobs=2; inv_num_jobs=0.5; scp_indexes=(pad 1 2); frame_shifts=(pad 1 2); dropout_opt="--edits='set-dropout-proportion name=* proportion=0.0'" lrate=0.002
+  . <(grep "^iter=$x;" $dir/schedule.txt)
+
+  echo "$0: training, iteration $x of $num_iters, num-jobs is $num_jobs"
+
+  next_x=$[$x+1]
+  den_fst_dir=$egs_dir/misc
+  model_out_prefix=$dir/${next_x}
+  model_out=${model_out_prefix}.mdl
+  multilingual_eg_opts=
+  if $multilingual_eg; then
+       multilingual_eg_opts="--multilingual-eg=true"
+  fi
+
+  # for the first 4 iterations, plus every $diagnostic_period iterations, launch
+  # some diagnostic processes.  We don't do this on iteration 0, because
+  # the batchnorm stats wouldn't be ready
+  if [ $x -gt 0 ] && [ $[x%diagnostic_period] -eq 0 -o $x -lt 5 ]; then
+
+    [ -f $dir/.error_diagnostic ] && rm $dir/.error_diagnostic
+    for name in train heldout; do
+      egs_opts=
+      if $multilingual_eg; then
+          weight_rspecifier=$egs_dir/diagnostic_${name}.weight.ark
+          [[ -f $weight_rspecifier ]] && egs_opts="--weights=ark:$weight_rspecifier"
+      fi
+      $cmd $gpu_cmd_opt $dir/log/diagnostic_${name}.$x.log \
+         nnet3-chain-train2 --use-gpu=$use_gpu \
+            --leaky-hmm-coefficient=$leaky_hmm_coefficient \
+            --xent-regularize=$xent_regularize \
+            --out-of-range-regularize=$out_of_range_regularize \
+            $l2_regularize_opt \
+            --print-interval=10  \
+           "nnet3-copy --learning-rate=$lrate $dir/${x}.raw - |" $den_fst_dir \
+           "ark:nnet3-chain-copy-egs $egs_opts scp:$egs_dir/${name}_subset.scp ark:- | nnet3-chain-merge-egs $multilingual_eg_opts --minibatch-size=1:64 ark:- ark:-|" \
```

Comment:
`--minibatch-size=1:64` was previously `--minibatch-size=$groups_per_minibatch`.  Was  hardcoding the value intentional?